### PR TITLE
Add syncing Mongo attributes after submitting changes to RTD 

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

[MTC configuration]
This PR changes the behavior of the MTC Feed Attributes editor as follows when the user finishes editing an MTC feed attribute:
- The new attribute value is directly submitted (PUT) to RTD (it is no longer written prior to sending to RTD)
- After the PUT call to RTD, a GET call is made to RTD to fetch attributes anew and write those attributes to Mongo:
    * Feed attributes in Mongo that are not sent by RTD (e.g.  `AgencyPublicId`) are removed from Mongo.
    * Feed attributes sent by RTD not in Mongo are added to Mongo.

The changes should ensure that RTD feed attributes are synced when editing feed attributes.

Suggestions for refactoring and adding tests welcome!

### CODE REVIEW

**Comment by binh-dam-ibigroup** [comment]
Ready for review again with unit tests added in https://github.com/ibi-group/datatools-server/pull/438/commits/be92d52c9a0eda524b485e6b7d39302b3e550781.

File: src/test/resources/com/conveyal/datatools/mtc-rtd-mock-responses/mtc-rtd-mock-responses.iml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (br648) [inline src/test/resources/com/conveyal/datatools/mtc-rtd-mock-responses/mtc-rtd-mock-responses.iml:1]: This file shouldn't be included.

**Review by br648 — APPROVED** [review]
LGTM apart from the IntelliJ modules file which should be removed.

File: src/test/resources/com/conveyal/datatools/mtc-rtd-mock-responses/mtc-rtd-mock-responses.iml (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/test/resources/com/conveyal/datatools/mtc-rtd-mock-responses/mtc-rtd-mock-responses.iml]: Thanks, removed in https://github.com/ibi-group/datatools-server/pull/438/commits/3fc8d79507c45e0debea70254c05cc34c4686535.

**Comment by daniel-heppner-ibigroup** [comment]
Looks good! Good for me to look at this code and get used to it.

**Review by daniel-heppner-ibigroup — APPROVED** [review]

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks great! Very clean fix

**Pull request merged**